pkg/querier/queryrange/split_by_interval.go

```diff
@@ -8,8 +8,10 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
+type IntervalFn func(r Request) time.Duration
```

Comment:
What are the requirements on this function? Can it really return different interval for each request? How will that interact with the caching?

Reply:
I'm curious about this as well. I'm happy to merge it, while we discuss this.